src/components/lib/cos_kernel_api.c

```diff
@@ -421,7 +422,7 @@ cos_captbl_alloc(struct cos_compinfo *ci)
 	assert(ci);
 
 	if (__alloc_mem_cap(ci, CAP_CAPTBL, &kmem, &cap)) return 0;
-	if (call_cap_op(ci->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, cap, ci->pgtbl_cap, kmem, 0)) BUG();
+	if (call_cap_op(ci->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, cap, __compinfo_metacap(ci)->local_mi.pgtbl_cap, kmem, 0)) BUG();
```

Comment:
`SELF_PT` is the offset in any captbl in my understanding. It wasn't literal meaning of "SELF" that I was referring to. So, `ci->pgtbl_cap` is `SELF_PT` of `ci`, same way for `meta`. 

But I understand now what you mean by `SELF_PT`. And I also understand that we are going to have physical memory mappings in `UNTYPED_PT` and virtual memory mappings in `SELF_PT`, which is what we mostly use `SELF_PT` for anyway. Physical memory pages are only alloced and mapped in `memsrc`, so it should not impact the API much, I believe.